examples/nlp/nmt_tutorial.py

```diff
@@ -225,7 +225,7 @@ def create_pipeline(dataset_src,
 
 # callback which saves checkpoints once in a while
 ckpt_callback = nemo.core.CheckpointCallback(
-    folder=nf.checkpoint_dir,
+    folder=args.checkpoint_dir,
```

Comment:
Please use nf.checkpoint_dir

Reply:
Either separate the checkpoint arg from the restore model dir or pass args.checkpoint_dir to the Neural Factory initializer. I find it very odd that this script combines both training and inference.

Reply:
Is it possible for nf.checkpoint_dir to point for specified <ckpt_dir> folder, or the checkpoints should always be at <ckpt_dir>/checkpoints?

Reply:
Here's the problem, if you don't pass the checkpoint_dir argument to the neural factory constructor. It will default to log_dir/checkpoint. Here's the problem if you're running this in training, you would want it to be in log_dir/checkpoint. Otherwise for inference, you would want it to be in checkpoint_dir.

Which is why I am confused as to why you are doing two things in one script.

Reply:
You want checkpointcallback to be doing two different things depending on what mode you are running your script it.

Also as it stands, checkpoint_dir is not defined in your argparser.